packages/core/admin/admin/src/components/Theme.tsx

```diff
@@ -10,9 +11,29 @@ interface ThemeProps {
   children: React.ReactNode;
 }
 
+type NonSystemThemeName = Exclude<ThemeName, 'system'>;
+
 const Theme = ({ children }: ThemeProps) => {
   const { currentTheme, themes } = useThemeToggle();
   const { locale } = useIntl();
+  const [systemTheme, setSystemTheme] = React.useState<NonSystemThemeName>();
+
+  // Listen to changes in the system theme
+  React.useEffect(() => {
+    const themeWatcher = window.matchMedia('(prefers-color-scheme: dark)');
+    setSystemTheme(themeWatcher.matches ? 'dark' : 'light');
+
+    themeWatcher.addEventListener('change', (event) => {
+      setSystemTheme(event.matches ? 'dark' : 'light');
+    });
+
+    // Cleanup on unmount
+    return () => {
+      themeWatcher.removeEventListener('change', () => {});
+    };
+  }, []);
+
+  const computedThemeName = currentTheme === 'system' ? systemTheme : currentTheme;
```

Comment:
I wonder if this is better off in the `ThemeToggleProvider` component? Interested to hear why you decided to put it in this component? :)

Reply:
I figured it was a theme concern, but it's even more of a theme toggle concern 😁